File: sqs/nodejs/sqs_client.js
var AWS = require('aws-sdk');
var fs = require('fs');

AWS.config.loadFromPath('/path/to/config.json');

var sqs = new AWS.SQS();

var params = {
  QueueUrl: 'https://sqs.us-west-2.amazonaws.com/xxxxx/your_queue',
  MessageBody: 'Send Message!'
}
sqs.sendMessage(params, function(err, data) {
  if (err)
    console.log(err);
  else
    console.log("Successfully sent.");
})


var params = {
  QueueUrl: 'https://sqs.us-west-2.amazonaws.com/xxxxx/your_queue',
}
sqs.receiveMessage(params, function(err, data) {
  console.log(data);
})

File: sqs/ruby/sqs_client.rb
#!/usr/bin/env ruby
require 'aws'
require 'pp'

AWS.config({
  :access_key_id => ENV['AWS_ACCESS_KEY_ID'],
  :secret_access_key => ENV['AWS_SECRET_ACCESS_KEY'],
  :region => ENV['AWS_REGION']
})

sqs = AWS::SQS.new

queue_url = 'https://sqs.us-west-2.amazonaws.com/xxxxx/your_queue'
result = sqs.client.send_message({
  :queue_url    => queue_url,
  :message_body => 'Send Message!'
})
pp result

result = sqs.client.receive_message({
  :queue_url    => queue_url
})
pp result
